pkg/gatherers/clusterconfig/events_filtering.go

```diff
@@ -0,0 +1,72 @@
+package clusterconfig
+
+import (
+	"sort"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+)
+
+//filterEvents() returns events that occoured since last interval
+func filterEvents(interval time.Duration, events *v1.EventList, Type string) v1.EventList {
+	oldestEventTime := time.Now().Add(-interval)
+	var filteredEvents = v1.EventList{}
+	switch Type {
+	case "Warning":
+		for i := range events.Items {
+			if isEventAbnormal(&events.Items[i]) && isEventNew(&events.Items[i], oldestEventTime) {
+				filteredEvents.Items = append(filteredEvents.Items, events.Items[i])
+			}
+		}
+	default:
+		for i := range events.Items {
+			if isEventNew(&events.Items[i], oldestEventTime) {
+				filteredEvents.Items = append(filteredEvents.Items, events.Items[i])
+			}
+		}
```

Comment:
I still don't like this switch much to be honest :) What if someone passes "Error" type for example? Perhaps I would split this function into two:
- `getEventsForInterval` (or from interval? Naming is difficult:-)) - this would handle just the time "filtering"
- `filterAbnormalEvents` or something like that and this function (or logic) can stay in the new gatherer for now